src/python/pants/jvm/jdk_rules.py

```diff
@@ -36,11 +37,41 @@ class Nailgun:
     classpath_entry: ClasspathEntry
 
 
+class DefaultJdk(Enum):
+    SYSTEM = "system"
+    SOURCE_DEFAULT = "source_default"
+
+
 @dataclass(frozen=True)
 class JdkRequest:
-    """Request for a JDK with a specific major version, or None (use System JDK)."""
+    """Request for a JDK with a specific major version, or a default (`--jvm-jdk` or System)."""
+
+    version: str | DefaultJdk
+
+    @classproperty
```

Comment:
Is there a `@staticproperty`?

Reply:
if there is, vscode couldn't find it

Reply:
I actually just finished deleting that. What would be the benefit? https://github.com/pantsbuild/pants/pull/14385

Reply:
Since `cls` is unused here. But not something I'd have held this up for.